### PR TITLE
Adds Chuunibyou Spell + Granter

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

My April fools this year, though not going to call it one because some people think it should just be actually merged.

### Chuunibyou Powers 🌟

Wizard gets a new spell for 2 points that gives him the powers of chuuni. This makes them have ridiculous shouted invocations for all their spells, their spells are colored pink, and they heal slightly when casting one.

While mostly a meme spell, I could see a tailored loadout like lichdom and splattercasting that takes advantage of the unique spellcasting changes, like a very low cooldown spammable loadout to heal quickly.

There is also a granter book in the library, which teaches a version of chunni that doesn't heal.

#### Medical eyepatch

I added it, chuuni wizards get a NODROP version.

## Why It's Good For The Game

This PR bestows upon the game the glorious gift of chuuni powers, the ultimate manifestation of my hidden potential and the secret truth of this world, which only I and a few chosen ones can comprehend and unleash! Why wouldn't you want it?!

In all seriousness, it is a unique wizard playstyle and it will make for some funny memes. Beyond wizard, the chaplain, heretics, or mime can read it in the library for a very silly round. I like it!

## Changelog
:cl:
add: Chuunibyou wizards, and chunni granters in the library
add: Medical eyepatches
/:cl:

### CODE REVIEW

**Comment by Cheshify** [comment]
This is it, the best PR

**Comment by Overuseddart** [comment]
You can probably buy multiple of the random teleport spell or buy rod 8 times so you can be constantly healing, honestly sounds really fun to try to kill healing rod wizard.

**Comment by tralezab** [comment]
> You can probably buy multiple of the random teleport spell or buy rod 8 times so you can be constantly healing, honestly sounds really fun to try to kill healing rod wizard.

I put a pretty low cooldown so you're still getting an advantage from casting often but you can't just instant blink and become almost immortal

**Comment by Shadowflame909** [comment]
Explosion!!!

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -200,6 +200,29 @@
...
 
+/obj/item/clothing/glasses/eyepatch/medical
+	name = "medical eyepatch"
+	desc = "Used by space weeaboos to pretend their eye isn't there, and crewmembers who actually lost their eye to pretend their eye is there."
+	icon_state = "eyepatch_medical"
+	base_icon_state = "eyepatch_medical"
+	inhand_icon_state = null
+
+/// wizard version
+/obj/item/clothing/glasses/eyepatch/medical/chuuni
```
Comment (ATH1909) [inline code/modules/clothing/glasses/_glasses.dm:211]: should the wizard version get flash protection so you don't end up being pushed to discard it in favor of sechuds in order to not instalose to borgs/flashbangs?

Comment (Fikou) [reply code/modules/clothing/glasses/_glasses.dm:211]: you cant discard it its nodrop

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -200,6 +200,29 @@
...
+
+/obj/item/clothing/glasses/eyepatch/medical/chuuni/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, type)
```
Comment (Fikou) [inline code/modules/clothing/glasses/_glasses.dm:215]: make it only become nodrop when you equip it

File: code/modules/library/random_books.dm [hunk 3]
```diff
@@ -71,12 +71,22 @@
 /obj/structure/bookcase/random/fiction
 	name = "bookcase (Fiction)"
 	random_category = "Fiction"
+	///have we spawned the chuuni granter
+	var/static/chuuni_book_spawned = FALSE
+
+/obj/structure/bookcase/random/fiction/after_random_load()
+	if(!chuuni_book_spawned)
```
Comment (Fikou) [inline code/modules/library/random_books.dm:78]: youd wanna check here if its the station level i presume

File: code/modules/spells/spell.dm [hunk 4]
```diff
@@ -65,6 +65,8 @@
 	var/invocation
 	/// What is shown in chat when the user casts the spell, only matters for INVOCATION_EMOTE
 	var/invocation_self_message
+	/// if true, doesn't garble the invocation sometimes with backticks
+	var/garbled_invocation = TRUE
```
Comment (Fikou) [inline code/modules/spells/spell.dm:69]: why not something like garble_chance and then you can just use prob(garble_chance), set it to 0 where you dont want it

Comment (tralezab) [reply code/modules/spells/spell.dm:69]: i like that

File: code/modules/clothing/glasses/_glasses.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/clothing/glasses/_glasses.dm:215]: sure, but the only time you'll see this is when the wizard has it plastered on his face

**Comment by CMDR-Gungnir** [comment]
I'm actually curious how it would interact with Mimery. Does the Mime do the chant and thus break their vow? Does the Mime not end up saying anything after all?

**Comment by AlamoTurtle** [comment]
> I'm actually curious how it would interact with Mimery. Does the Mime do the chant and thus break their vow? Does the Mime not end up saying anything after all?

Maybe, but it might also apply to the ventriloquist dummy.

**Comment by tralezab** [comment]
> I'm actually curious how it would interact with Mimery. Does the Mime do the chant and thus break their vow? Does the Mime not end up saying anything after all?

Before, it gave an error about not speaking but cast the spell anyway. Now, it makes the mime do the incantation, even if it gets them lynched.
image
And I like how they talk like a mime while casting too. 
image

**Pull request merged**